xbmc/video/VideoDatabase.cpp

```diff
@@ -12340,21 +12340,23 @@ bool CVideoDatabase::GetVideoVersionsNav(const std::string& strBaseDir,
     std::string strSQL;
 
     if (idMedia != -1)
-      strSQL = PrepareSQL("SELECT videoversiontype.name AS name,"
-                          "  videoversiontype.id AS id "
-                          "FROM videoversiontype"
-                          "  JOIN videoversion ON"
-                          "    videoversion.idType = videoversiontype.id "
-                          "WHERE idMedia = %i AND mediaType = '%s'",
-                          idMedia, mediaType.c_str());
+      strSQL =
+          PrepareSQL("SELECT videoversiontype.name AS name,"
+                     "  videoversiontype.id AS id "
+                     "FROM videoversiontype"
+                     "  JOIN videoversion ON"
+                     "    videoversion.idType = videoversiontype.id "
+                     "WHERE idMedia = %i AND mediaType = '%s' AND videoversiontype.itemType = %i",
+                     idMedia, mediaType.c_str(), VideoAssetType::VERSION);
     else
-      strSQL = PrepareSQL("SELECT DISTINCT videoversiontype.name AS name,"
-                          "  videoversiontype.id AS id "
-                          "FROM videoversiontype"
-                          "  JOIN videoversion ON"
-                          "    videoversion.idType = videoversiontype.id "
-                          "WHERE name != '' AND owner IN (%i, %i)",
-                          VideoAssetTypeOwner::SYSTEM, VideoAssetTypeOwner::USER);
+      strSQL = PrepareSQL(
+          "SELECT DISTINCT videoversiontype.name AS name,"
+          "  videoversiontype.id AS id "
+          "FROM videoversiontype"
+          "  JOIN videoversion ON"
+          "    videoversion.idType = videoversiontype.id "
+          "WHERE name != '' AND owner IN (%i, %i) AND videoversiontype.itemType = %i",
+          VideoAssetTypeOwner::SYSTEM, VideoAssetTypeOwner::USER, VideoAssetType::VERSION);
```

Comment:
SQL is ok, but I haven't grasped the navigation concepts yet and in particular what it means for versions / extras.
Isn't there a similar nav for extras? In which case hardcoding VideoAssetType::VERSION is not great.

Reply:
> Isn't there a similar nav for extras? In which case hardcoding VideoAssetType::VERSION is not great.

No, not yet. And mixing extras and versions is exactly what we want to prevent. So, for now the hard-coded type is okay.

Reply:
ok